### PR TITLE
apiserver: use the correct error when logging errors updating managedFields

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <andrewsy@google.com>

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

I noticed that `[SHOULD NOT HAPPEN] failed to update managedFields` was being logged but there was no error logged with it. After looking at the code, it's because the err passed into `klog.ErrorS` is the nil error from `meta.Accessor(newObj)` when it should be the err from trying to update `managedFields`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
apiserver: use the correct error when logging errors updating managedFields
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go [hunk 1]
```diff
@@ -202,8 +202,8 @@ func (f *FieldManager) UpdateNoErrors(liveObj, newObj runtime.Object, manager st
 	if err != nil {
 		atMostEverySecond.Do(func() {
 			ns, name := "unknown", "unknown"
-			accessor, err := meta.Accessor(newObj)
-			if err == nil {
+			accessor, accessorErr := meta.Accessor(newObj)
```
Comment (andrewsykim) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:205]: Otherwise error log on line 211 uses a nil err

**Review by andrewsykim — COMMENTED** [review]
/assign @lavalamp 

Also generally curious if we should be concerned that this is logging

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go [hunk 2]
```diff
@@ -202,8 +202,8 @@ func (f *FieldManager) UpdateNoErrors(liveObj, newObj runtime.Object, manager st
...
+			if accessorErr == nil {
 				ns = accessor.GetNamespace()
 				name = accessor.GetName()
 			}
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:209]: Wouldn't it be cuter to do:
```suggestion
			if accessor, err := meta.Accessor(newObj); err == nil {
				ns = accessor.GetNamespace()
				name = accessor.GetName()
			}
```

**Comment by apelisse** [comment]
> Also generally curious if we should be concerned that this is logging

I guess we'll only be able to tell once we see the error :-)

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go (hunk 2 above)
Comment (lavalamp) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:209]: +1

Comment (andrewsykim) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go]: Updated, thanks

**Pull request merged**